lua/plenary/path.lua

```diff
@@ -314,6 +314,47 @@ function Path:rmdir()
   uv.fs_rmdir(self:absolute())
 end
 
+function Path:rename(opts)
+  opts = opts or {}
+  if not opts.new_name or opts.new_name == "" then
+    print("Please provide the new name!")
+    return false
+  end
+
+  -- handles `.`, `..`, `./`, and `../`
+  if opts.new_name:match('^%.%.?/?') then
+    print("Invalid filename!")
```

Comment:
We should probably use `error` i am not sure about it tho. I need to think about it

Reply:
At least we are using `error` everywhere. You can catch it with `pcall`

Reply:
yeah, I think `error` is better.